ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadWork.java

```diff
@@ -61,12 +63,19 @@
   */
   final LineageState sessionStateLineageState;
 
-  public ReplLoadWork(HiveConf hiveConf, String dumpDirectory, String dbNameToLoadIn,
-      LineageState lineageState, boolean isIncrementalDump, Long eventTo,
-      List<DirCopyWork> pathsToCopyIterator) throws IOException {
+  public ReplLoadWork(HiveConf hiveConf, String dumpDirectory,
+                      String dbNameToLoadIn, ReplScope changedReplScope,
+                      LineageState lineageState, boolean isIncrementalDump, Long eventTo,
+                      List<DirCopyWork> pathsToCopyIterator) throws IOException {
     sessionStateLineageState = lineageState;
     this.dumpDirectory = dumpDirectory;
     this.dbNameToLoadIn = dbNameToLoadIn;
+    this.changedReplScope = changedReplScope;
+
+    // If DB name is changed during REPL LOAD, then set it instead of referring to source DB name.
+    if ((changedReplScope != null) && StringUtils.isNotBlank(dbNameToLoadIn)) {
```

Comment:
Yes, dump metadata will always have DB name at source. If user explicitly mentioned db name in REPL LOAD, then dbNameToLoadIn will not be blank. Also, we don't pass real policy in dump metadata always. Only if the policy changes, then only passed. This is an optimization to avoid traversing all tables. So, changedReplScope can be null and also dbNameToLoadIn can be blank or not blank.